### DISCUSSION TITLE
Get rid of System.Drawing.Common for Svg.Custom

### DISCUSSION THREAD

**Discussion by Happypig375** [wieslawsoltes/Svg.Skia#93] [Ideas]
https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only

Even though Svg.Skia should use SkiaSharp instead of System.Drawing.Common everywhere, System.Drawing.Common is still a dependency through Svg.Custom. It should not be used anymore.

**Comment by wieslawsoltes** [comment]
This library doesn't target only .NET 6

**Reply by Happypig375** [reply]
For going forward, System.Drawing.Common may blow up in .NET 7+. Anyways, SkiaSharp should be used wherever possible.

**Reply by wieslawsoltes** [reply]
Do not blame me for Microsoft not developing System.Drawing.Common for cross-platform. Also note SkiaSharp may also blow up sooner then later, read this https://github.com/mono/SkiaSharp/discussions/1884

The only viable solution is using native Skia from C++ but it breaks evry new release made by Goodle. Good luck with that too.

**Comment by wieslawsoltes** [comment]
@Happypig375 Done in here https://github.com/wieslawsoltes/Svg.Skia/pull/94

**Comment by wieslawsoltes** [comment]
There are now 902 build errors. This will take time to implement properly.

**Reply by Happypig375** [reply]
They should be from Svg.Custom which does expose System.Drawing.Common types.

**Reply by wieslawsoltes** [reply]
Now we need to substitute those types with something else:
- ImageSharp
- SkiaSharp
- Microsoft.Maui.Graphics

but this takes new dependency.

**Reply by wieslawsoltes** [reply]
> They should be from Svg.Custom which does expose System.Drawing.Common types.

Also now I need to forks SVG library and remove all dependencies from  System.Drawing.Common types where they are used, like everywhere, this will break all backwards compatibility as even core interfaces are dependent on  System.Drawing.Common types. So now I have to maintain custom fork of SVG and merge manually all changes from upstream.

**Comment by wieslawsoltes** [comment]
@Happypig375 You can try https://github.com/wieslawsoltes/Svg.Skia/pull/94

**Comment by wieslawsoltes** [comment]
@Happypig375 The PR has been merged into master branch so no more System.Drawing.Common

**Comment by wieslawsoltes** [comment]
Released version 0.5.11.1 without System.Drawing.Common dependency!

**Reply by Happypig375** [reply]
Thank you for your quick response!

**Reply by Happypig375** [reply]
However, now interfaces like ISvgBoundable and ISvgClippable are now empty. Can SkiaSharp be used to calculate those properties instead?

**Reply by wieslawsoltes** [reply]
I have alternative model derived from base Svg model that has all of those properties. No plans to port that back as it's way to complex already.

**Reply by wieslawsoltes** [reply]
https://github.com/wieslawsoltes/Svg.Skia/tree/master/src/Svg.Model/Drawables

**Reply by wieslawsoltes** [reply]
https://github.com/wieslawsoltes/Svg.Skia/blob/f9808bd236ff2f4db75db0bd636b39d684f72366/src/Svg.Model/Drawables/DrawableBase.cs#L23-L24

**Reply by wieslawsoltes** [reply]
https://github.com/wieslawsoltes/Svg.Skia/blob/f9808bd236ff2f4db75db0bd636b39d684f72366/src/Svg.Skia/SKSvg.cs#L15

**Reply by Happypig375** [reply]
How can I use ToModel on an arbitrary SvgElement (not necessarily SvgFragment) that is looked up by Id?

**Reply by Happypig375** [reply]
Drawables don't have Ids attached

**Reply by Happypig375** [reply]
Nevermind, each drawable references the element where I can get ID from. However, for text, the drawable version provides zero bounds while the element version provides non-zero bounds. How can I get bounds from text?

**Reply by Happypig375** [reply]
For rectangles with transforms, drawable TransformedBounds have slightly different values than element Bounds, like differences of 0.5 to 1. Which one is more accurate?

**Reply by Happypig375** [reply]
Btw, you can also remove Svg.SvgDocument.SystemIsGdiPlusCapable, Svg.SvgDocument.EnsureSystemIsGdiPlusCapable, Svg.SvgDocument.ExceptionCaughtIsGdiPlusRelated, and Svg.SvgDocument.SkipGdiPlusCapabilityCheck since they don't do anything anymore.

**Reply by wieslawsoltes** [reply]
@Happypig375 Will do in https://github.com/svg-net/SVG/pull/934

**Comment by wieslawsoltes** [comment]
> [docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only](https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only)
> 
> Even though Svg.Skia should use SkiaSharp instead of System.Drawing.Common everywhere, System.Drawing.Common is still a dependency through Svg.Custom. It should not be used anymore.

@Happypig375 I hope you did read the disclaimer of one of the recommended libraries https://github.com/dotnet/Microsoft.Maui.Graphics#disclaimer

**Reply by Happypig375** [reply]
Uh, do you mean that there is no support, use at own risk, for this library?

**Reply by Happypig375** [reply]
It's placed last after SkiaSharp though.

**Reply by wieslawsoltes** [reply]
I think both SkiaSharp and Microsoft.Maui.Graphics come with no guaranties for support.
